notebooks/data_oob.ipynb

```diff
@@ -43,167 +43,198 @@
+       "3   53  234721              7             0             0              40   \n",
+       "4   28  338409             13             0             0              40   \n",
+       "\n",
+       "  income  \n",
+       "0  <=50K  \n",
+       "1  <=50K  \n",
+       "2  <=50K  \n",
+       "3  <=50K  \n",
+       "4  <=50K  "
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
     }
    ],
    "source": [
-    "import pandas as pd\n",
-    "\n",
-    "# Specify the URL of the dataset\n",
-    "url = \"https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data\"\n",
-    "\n",
-    "# Specify the column names\n",
-    "column_names = [\n",
-    "    \"age\",\n",
-    "    \"workclass\",\n",
-    "    \"fnlwgt\",\n",
-    "    \"education\",\n",
-    "    \"education-num\",\n",
-    "    \"marital-status\",\n",
-    "    \"occupation\",\n",
-    "    \"relationship\",\n",
-    "    \"race\",\n",
-    "    \"sex\",\n",
-    "    \"capital-gain\",\n",
-    "    \"capital-loss\",\n",
-    "    \"hours-per-week\",\n",
-    "    \"native-country\",\n",
-    "    \"income\",\n",
-    "]\n",
-    "\n",
-    "# Specify the data types for each column\n",
-    "data_types = {\n",
-    "    \"age\": int,\n",
-    "    \"workclass\": \"category\",\n",
-    "    \"fnlwgt\": int,\n",
-    "    \"education\": \"category\",\n",
-    "    \"education-num\": int,\n",
-    "    \"marital-status\": \"category\",\n",
-    "    \"occupation\": \"category\",\n",
-    "    \"relationship\": \"category\",\n",
-    "    \"race\": \"category\",\n",
-    "    \"sex\": \"category\",\n",
-    "    \"capital-gain\": int,\n",
-    "    \"capital-loss\": int,\n",
-    "    \"hours-per-week\": int,\n",
-    "    \"native-country\": \"category\",\n",
-    "    \"income\": \"category\",\n",
-    "}\n",
-    "\n",
-    "# Load the dataset into a pandas DataFrame\n",
-    "data_adult = pd.read_csv(\n",
-    "    url,\n",
-    "    names=column_names,\n",
-    "    sep=\",\\s*\",\n",
-    "    engine=\"python\",\n",
-    "    na_values=\"?\",\n",
-    "    dtype=data_types,\n",
-    "    nrows=2000,\n",
-    ")\n",
+    "data_adult = load_adult_data()\n",
+    "data_adult.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Computing the OOB values\n",
     "\n",
-    "# Drop categorical columns\n",
-    "data_adult = data_adult.drop(\n",
-    "    columns=[\n",
-    "        \"workclass\",\n",
-    "        \"education\",\n",
-    "        \"marital-status\",\n",
-    "        \"occupation\",\n",
-    "        \"relationship\",\n",
-    "        \"race\",\n",
-    "        \"sex\",\n",
-    "        \"native-country\",\n",
-    "    ]\n",
-    ")\n",
+    "The main idea of Data-OOB is to take a pre-trained classifier and compute a per-sample out-of-bag performance estimate via bagging.\n",
```

Comment:
I think there's a mistake here. It says that we're taking a pre-trained classifier but in the code and the method we trained one from scratch using bagging.

Reply:
Hmm... you are right, it's not "pre-trained", but "existing".